tests/hotplug/instancetype.go

```diff
@@ -182,4 +184,111 @@ var _ = Describe("[sig-compute]Instance Type and Preference Hotplug", decorators
+				const memoryPath = "/spec/template/spec/domain/resources/requests/memory"
+				patches := patch.New(
+					patch.WithTest(memoryPath, vm.Spec.Template.Spec.Domain.Resources.Requests.Memory().String()),
+					patch.WithReplace(memoryPath, newMemory.String()),
+				)
+				patchData, err := patches.GeneratePayload()
+				Expect(err).NotTo(HaveOccurred())
+				return virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
+			},
+			ContainSubstring("failure checking preference requirements: insufficient Memory resources of 512Mi provided by VirtualMachine, preference requires 1Gi"),
+		),
+		Entry("by instance type resource requests",
+			func(vmi *virtv1.VirtualMachineInstance, preferenceName string) (*virtv1.VirtualMachine, error) {
+				origInstancetype, err := virtClient.VirtualMachineInstancetype(vmi.Namespace).Create(
+					context.Background(),
+					builder.NewInstancetypeFromVMI(vmi),
+					metav1.CreateOptions{},
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				vm, err := virtClient.VirtualMachine(vmi.Namespace).Create(
+					context.Background(),
+					libvmi.NewVirtualMachine(
+						vmi,
+						libvmi.WithInstancetype(origInstancetype.Name),
+						libvmi.WithPreference(preferenceName),
+					),
+					metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				// bug #14595 was caused by lookup code using the stashed controller
+				// revision copy of a different object during live update so wait until
+				// these are created before asserting that we fail to live update
+				By("Waiting for VirtualMachineInstancetype and VirtualMachinePreference ControllerRevision to be referenced from the VirtualMachine")
+				Eventually(ThisVM(vm)).WithTimeout(1 * time.Minute).WithPolling(2 * time.Second).Should(HaveControllerRevisionRefs())
+				return vm, nil
+			},
+			func(vm *virtv1.VirtualMachine) (*virtv1.VirtualMachine, error) {
+				origInstancetype, err := virtClient.VirtualMachineInstancetype(vm.Namespace).Get(
+					context.Background(), vm.Spec.Instancetype.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				newMemory := origInstancetype.Spec.Memory.Guest.DeepCopy()
+				newMemory.Sub(resource.MustParse("512Mi"))
+				newInstancetype, err := virtClient.VirtualMachineInstancetype(vm.Namespace).Create(
+					context.Background(),
+					builder.NewInstancetype(
+						builder.WithCPUs(1),
+						builder.WithMemory(newMemory.String()),
+					),
+					metav1.CreateOptions{},
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				patches := patch.New(
+					patch.WithTest("/spec/instancetype/name", origInstancetype.Name),
+					patch.WithReplace("/spec/instancetype/name", newInstancetype.Name),
+				)
+				patchData, err := patches.GeneratePayload()
+				Expect(err).NotTo(HaveOccurred())
+				return virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
```

Comment:
Do we need to wait for updated CR refs to avoid flakes?

Reply:
No, this patch will always fail in the webhook now so the updated VirtualMachine is never seen by the controller or sync loop.